src/ns/core/notifier.py

```diff
@@ -69,11 +72,18 @@ def _construct_email(
         try:
             plaintext_email = plaintext_template.substitute(payload_as_dict)
         except KeyError as err:
-            raise self.VariableNotSuppliedError(variable=err.args[0]) from err
+            template_var_error = self.VariableNotSuppliedError(variable=err.args[0])
+            log.critical(template_var_error, extra={"variable": err.args[0]})
+            raise template_var_error from err
         except ValueError as err:
-            raise self.BadTemplateFormat(
+            template_format_error = self.BadTemplateFormat(
                 template_type="plaintext", problem=err.args[0]
```

Comment:
Not strictly necessary, but this might benefit from using an enum for the template_type, constraining the possible values + having one place where the actual string value is defined

Reply:
@mephenor see the changes in [c200c44](https://github.com/ghga-de/notification-service/pull/10/commits/c200c447efe5160fe557568c38ba00c60a9edfb9)